[9.0][ADD] web_widget_float_highlight

espo-tony opened the pull request:
This module extends the functionality of web backend to allow user to highlight
any numeric field when its value belongs to a given interval.

aesposito-onestein committed:
[9.0][ADD] web_widget_float_highlight

espo-tony commented:
For those interested in seeing the features of this module at work, an example of implementation can be found [here](https://github.com/onesteinbv/addons-onestein/tree/9.0/account_activity_based_costing)

tarteo commented on web_widget_float_highlight/static/src/js/web_highlight.js:
```diff
@@ -0,0 +1,72 @@
+
+        init: function () {
+            this._super.apply(this, arguments);
+            var options = {
```
Consider using _.defaults

tarteo commented on web_widget_float_highlight/static/src/js/web_highlight.js:
```diff
@@ -0,0 +1,72 @@
+                always_work: (this.options && this.options.always_work) || false,
+                highlight: (this.options && this.options.highlight) || false
+            };
+            this.lower_threshold = options.lower_threshold;
```
Are those declarations necessary? Why don't you use this.options throughout the widget?

espo-tony replied:
It's not necessary but I really think it's desirable. As you can read, I'm not merely assigning the this.options value to the 'options' dictionary: I'm doing a selective assignment. If I missed the declaration of this dictionary, I should write much more code (much less readable and much harder to mantain) everytime I use these options.

tarteo commented on web_widget_float_highlight/static/src/js/web_highlight.js:
```diff
@@ -0,0 +1,72 @@
+        }
+    });
+
+
```
Too many new lines.

tarteo commented on web_widget_float_highlight/static/src/js/web_highlight.js:
```diff
@@ -0,0 +1,72 @@
+                lower_bg_color: (this.options && this.options.lower_bg_color) || "red",
+                middle_bg_color: (this.options && this.options.middle_bg_color) || "white",
+                upper_bg_color: (this.options && this.options.upper_bg_color) || "green",
+                lower_txt_color: (this.options && this.options.lower_txt_color) || "white",
```
In my opinion lower_font_color is a more appropriate name for this option.

tarteo commented on web_widget_float_highlight/static/src/js/web_highlight.js:
```diff
@@ -0,0 +1,72 @@
+        render_value: function () {
+            this._super();
+            var self = this;
+            if (self.highlight && (this.get("effective_readonly") || self.always_work)) {
```
Can you make the highlight option not required,  so that it is only used to set the default values for the other options?

tarteo commented on web_widget_float_highlight/static/src/js/web_highlight.js:
```diff
@@ -0,0 +1,72 @@
+                var val = this.get("value");
+                if (self.lower_threshold <= self.upper_threshold) {
+                    if (val < self.lower_threshold) {
+                        bg_color = self.lower_bg_color || "white";
```
You already set default values in the init function.

aesposito-onestein committed:
[IMP] Refactoring of the code and implemented a new way to use the widget without default values.

astirpe commented on web_widget_float_highlight/__openerp__.py:
```diff
@@ -0,0 +1,20 @@
+    'name': 'Highlight Float Field',
+    'images': [],
+    'category': 'Extra Tools',
+    'version': '10.0.1.0.0',
```
Sorry but I don't agree with this definition of version... too early to write "10".

lasley committed:
[ADD] web_responsive: Create new module
* Create App Drawer for primary menu
* Move secondary menu to top bar
* Add keyboard shortcuts for primary menu navigation
* Make web client slightly more mobile compatible (no widgets)

aesposito-onestein committed:
[FIX] Fixed version number in manifest

yajo commented on web_widget_float_highlight/static/src/js/web_highlight.js:
```diff
@@ -0,0 +1,66 @@
+                upper_bg_color: (this.options && this.options.upper_bg_color) || load_defaults && "green" || "white",
+                lower_font_color: (this.options && this.options.lower_font_color) || load_defaults && "white" || "#666666",
+                middle_font_color: (this.options && this.options.middle_font_color) || "#666666",
+                upper_font_color: (this.options && this.options.upper_font_color) || load_defaults && "white" || "#666666",
```
Can't you use bootstrap status classes instead?
http://getbootstrap.com/css/#helper-classes-colors

pedrobaeza commented:
https://github.com/OCA/web/pull/398 seems to be more powerful

aesposito-onestein committed:
[IMP] Implemented highlight functionality also on pivot views.

espo-tony commented:
With the latest commit, highlight functionality was also implemented on pivot views.

aesposito-onestein committed:
[FIX] Solved compatibility issue in highlight functionality on pivot view

yajo committed:
[FIX] web_responsive: Fix several issues
* one2many tags now not hidden behind inputs.
* Drawer icons now with much better look.
* No overflow bug in small views when closing drawer.
* Documented it when resizing viewport. (+1 squashed commit)
* Whitespace cleanup.

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

pedrobaeza committed:
[MIG] Remove setup module directories

pedrobaeza committed:
[MIG] Update metafiles

Commit:
[MIG] Migrated web_environment_ribbon to v10.0

pedrobaeza committed:
Merge pull request #436 from syleam/10.0-migrate-web_environment_ribbon

[MIG] Migrated web_environment_ribbon to v10.0

OCA-git-bot committed:
[ADD] setup.py

lasley committed:
[FIX] web_responsive: Other minor fixes
* Add Tecnativa as author
* Remove untestable JS method
* Rename to web_responsive

Commit:
Migration of web_widget_color to 10.0 (#439)

[MIG] web_widget_color: Migration to 10.0

pedrobaeza committed:
[REM] web_color: Overrided by web_widget_color

OCA-git-bot committed:
[ADD] setup.py

lasley committed:
[MIG] web_responsive: Migrate to v10
* Bump version
* Change all openerp to odoo
* Fix qunit suite injection
* Remove jQuery require
* Change to new selectors:
  * `oe_leftbar` to `o_sub_menu`
  * `oe_logo` to `o_sub_menu_logo`
  * `oe_footer` to `o_sub_menu_footer`
  * `oe_secondary_menus_container` to `o_sub_menu_content`
* Add style to hide oe_footer instead of remove to not break `support_branding`
* Add note in ReadMe explaining override of `support_branding`
* Set top margin of app drawer title to 0 to fit v10 proportions
* Fix notification badge positioning
* Add o_web_client class to body to fix overlaps
* Scroll control panel with page
* Change navbar z-index to not be overlapped by buttons
* Raise z-index on header to raise over buttons
* Handle layered notifications via z-index
* Remove `#` from navigation links HREF to prevent history littering

lasley committed:
[IMP] travis: Upgrade travis config
* Switch to new matrix
* Add Transifex
* Add travis_after_tests_success

pedrobaeza committed:
Merge pull request #448 from laslabs/hotfix/10.0/travis

[IMP][10.0] travis: Upgrade travis config

legalsylvain committed:
[REF] remove web_easy_switch_company. feature is now in the Odoo web Module

pedrobaeza committed:
Merge pull request #450 from legalsylvain/10.0_REF_remove_web_easy_switch_company

[REF] remove web_easy_switch_company. feature is now in the Odoo web module

legalsylvain committed:
10.0 port web sheet full width (#449)

[MIG] adapt web_sheet_full_width to V10.0 web framework

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] addons table in README.md

nikiwaibel committed:
[MIG] web_favicon: Migrated to 10.0 (#459)

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

lasley committed:
[FIX] web_responsive: Change z-indexes
* Rebalance z-indexes to fix overlay issue while still providing notification support

Commit:
Initial commit

Commit:
[IMP] Improved mixin name

Commit:
[ADD] Delete all button

Commit:
[ADD] README, copyrights, EOFs

Commit:
[FIX] xml declaration

Commit:
[FIX] Flake8

pedrobaeza committed:
Merge pull request #443 from laslabs/release/10.0/web_responsive

[MIG][10.0] web_responsive

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] addons table in README.md

danieltorres7 committed:
[10.0] web_tree_image: Porting to 10.0 (#464)

Commit:
[ADD] Tooltip on delete all button

moylop260 committed:
[ADD] web_x2many_delete_all (#477)

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

JayVora-SerpentCS committed:
[MIG] Migration started web_notify

JayVora-SerpentCS committed:
[MIG] Migration of  web_notify

JayVora-SerpentCS committed:
[MIG] Migration of  web_notify

jlzhou committed:
[MIG 10.0] web_export_view (#444)

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

yajo committed:
Expose mixin for inheritance, fix some spacing.

Related to https://github.com/OCA/web/pull/481.

Commit:
[ADD] Add setup.py

JayVora-SerpentCS committed:
Merge pull request #1 from acsone/10.0-setup-ape

[ADD] Add setup.py

Commit:
[FIX] Return result of super of show_application

JayVora-SerpentCS committed:
Update web_client.js

JayVora-SerpentCS committed:
Update __manifest__.py

oca-transbot committed:
OCA Transbot updated translations from Transifex

JayVora-SerpentCS committed:
Update __manifest__.py

JayVora-SerpentCS committed:
Merge pull request #2 from acsone/10.0-web-notify-return-ape

[FIX] Return result of super of show_application

sbidoul committed:
Merge pull request #491 from JayVora-SerpentCS/10-web-notify

Migrated web notify

Commit:
[MIG] web_send_message_popup 10.0 migration (#504)

* Intitial manifest changes

* [FIX] small README tricks

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

Commit:
[10.0] Add module web_no_bubble

pedrobaeza committed:
Merge pull request #509 from savoirfairelinux/10.0-web_no_bubble

[10.0] Add module web_no_bubble

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

danieltorres7 committed:
fixup! [MIG] web_responsive: Migrate to v10

Scroll control panel with page using CSS instead of jQuery.

jQuery scrolling caused scrollbar malfunction.

lasley committed:
Merge pull request #511 from danieltorres7/web_responsive_scroll_controlpanel

Scroll control panel with page using CSS instead of jQuery.

danieltorres7 committed:
Form view improvements to fit smaller screens

On screens with less than 768px content is limited to 2 columns.

On mobile field labels have their own line.

Fix elements width for XS screens

danieltorres7 committed:
Form image editing controls always present

Make image editing controls always available, instead of depending
on resolution or hover.

oca-transbot committed:
OCA Transbot updated translations from Transifex

danieltorres7 committed:
Web responsive menu improvements (#513)

* Bold titles for apps in the app-drawer
* Remove z-index from Control Panel buttons, so it doesn't overlap the menu.
* Better responsive menus
  * Move systray icons/menus out of the menu and up to the top bar.
  * Increase avatar size and visibility.

* Other minor menu behavior fixes.
* Changes to fit odoo variables
* And a little margin adjust on navbar-right
* Top menu always thick
* Removed overflow:hidden!important from main.less
* Local patch: prevent body overflow auto
* Stop using `style=` on `<body>`
* Fix jquery drawer in anticipation of:
   https://github.com/blivesta/drawer/pull/36

yajo committed:
Fix problem with overflowing chatter widget.

pedrobaeza committed:
Merge pull request #514 from danieltorres7/web_responsive_form_line_breaks

Web responsive form line breaks

Commit:
Initial commit

Commit:
[IMP] Improved description

Commit:
[IMP] Module's name

Commit:
[REM] Old folder

yajo committed:
[FIX][web_responsive] Form padding, titles and buttons OK in XS

Small makeup to make it look good in XS screens.

pedrobaeza committed:
Merge pull request #533 from Tecnativa/10.0-web_responsive-fix_titles_buttons

[FIX][web_responsive] Form padding, titles and buttons OK in XS

Commit:
[ADD] Selecting multiple ranges

Commit:
[IMP] Description for functionality

Commit:
[FIX] Copyright year

legalsylvain committed:
Merge pull request #532 from tarteo/100_add_web_listview_multi_select

[10.0][ADD][web_listview_range_select] Select a range of records with the shift key

espo-tony committed:
Merge remote-tracking branch 'remotes/upstream/10.0' into 90_add_web_widget_float_highlight
